src/block_entity.zig

```diff
@@ -175,9 +179,13 @@ pub const BlockEntityTypes = struct {
 		}
 
 		pub fn onLoadClient(_: Vec3i, _: *Chunk) void {}
-		pub fn onUnloadClient(_: Vec3i, _: *Chunk) void {}
+		pub fn onUnloadClient(_: BlockEntityIndex) void {}
 		pub fn onLoadServer(_: Vec3i, _: *Chunk) void {}
```

Comment:
Why inconsistent interface?

Reply:
loading needs to know where it is, in case the block entity makes use of the position (e.g. to render something at the position, or maybe spawn particles or whatever), in which it has to store it in storage
Furthermore the chunk is needed because we need to add the index to the chunk hashmap.